sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -68,20 +72,37 @@ export class AppConfigurationClient {
    * Initializes a new instance of the AppConfigurationClient class.
    * @param connectionString Connection string needed for a client to connect to Azure.
    */
-  constructor(connectionString: string) {
-    const regexMatch = connectionString.match(ConnectionStringRegex);
-    if (regexMatch) {
-      const appConfigCredential = new AppConfigCredential(regexMatch[2], regexMatch[3]);
-      this.client = new AppConfiguration(appConfigCredential, apiVersion, {
-        baseUri: regexMatch[1],
+  constructor(connectionString: string);
+  constructor(tokenCredential: TokenCredential, endpoint: string);
+  constructor(
+    connectionStringOrTokenCredential: string | TokenCredential,
+    endpoint?: string
+  ) {
+    if (isTokenCredential(connectionStringOrTokenCredential)) {
+      this.client = new AppConfiguration(connectionStringOrTokenCredential, apiVersion, {
+        baseUri: endpoint,
         deserializationContentTypes,
         requestPolicyFactories: (defaults) => [tracingPolicy(), ...defaults]
       });
     } else {
-      throw new Error("You must provide a connection string.");
+      const regexMatch = connectionStringOrTokenCredential.match(ConnectionStringRegex);
+      if (regexMatch) {
+        const appConfigCredential = new AppConfigCredential(regexMatch[2], regexMatch[3]);
+
+        this.client = new AppConfiguration(appConfigCredential, apiVersion, {
+          baseUri: regexMatch[1],
+          deserializationContentTypes,
+          requestPolicyFactories: (defaults) => [tracingPolicy(), ...defaults]
+        });
+      } else {
+        throw new Error("You must provide a connection string.");
```

Comment:
Hmm, should we say something like `Invalid connection string. Connection strings must be of the format "..."`?

Reply:
Yes. I printed out the regex.source which was clean enough.